build.rs

```diff
@@ -0,0 +1,214 @@
+    registry.is_valid()?;
+
+    let mut registry = registry.registry;
+    // The names are assumed to be sorted.
+    registry.sort_by_key(|a| a.name());
+
+    // Variables to insert into quote template:
+    let identifier: Vec<_> = registry
+        .iter()
+        .map(|r| format_ident!("{}", r.name()))
+        .collect();
+
+    let reserved_identifiers = registry
+        .iter()
+        .filter(|r| r.is_reserved())
+        .map(|r| format_ident!("{}", r.name()));
+
+    let count = registry.len();
+    let number: Vec<_> = registry.iter().map(|r| r.prefix).collect();
+    let enumeration: Vec<_> = (0..registry.len()).collect();
+
+    let mut prefix_to_idx: Vec<_> = number.iter().zip(enumeration).collect();
+    prefix_to_idx.sort_by_key(|(prefix, _)| *prefix);
+    let prefix_to_idx = prefix_to_idx
+        .iter()
+        .map(|(prefix, idx)| quote! { (#prefix, #idx) });
+
+    let name: Vec<_> = registry.iter().map(|r| r.network.clone()).collect();
+    let desc = registry.iter().map(|r| {
+        if let Some(website) = &r.website {
+            format!("{} - <{}>", r.display_name, website)
+        } else {
+            r.display_name.clone()
+        }
+    });
+
+    Ok(quote! {
+
+        /// A known address (sub)format/network ID for SS58.
+        #[non_exhaustive]
+        #[repr(u16)]
+        #[derive(Copy, Clone, PartialEq, Eq, crate::RuntimeDebug)]
+        pub enum KnownSs58AddressFormat {
+            #(#[doc = #desc] #identifier = #number),*,
+        }
+
+        /// All non-custom address formats (Sorted by name)
+        static ALL_SS58_ADDRESS_FORMATS: [KnownSs58AddressFormat; #count] = [
+             #(KnownSs58AddressFormat::#identifier),*,
+        ];
+
+        /// Names of all address formats (Sorted by name)
+        static ALL_SS58_ADDRESS_FORMAT_NAMES: [&str; #count] = [
+            #(#name),*,
+        ];
+
+        /// (Sorted) prefixes to index of ALL_SS58_ADDRESS_FORMATS
+        static PREFIX_TO_INDEX: [(u16, usize); #count] = [
+            #(#prefix_to_idx),*,
+        ];
+
+        impl Ss58AddressFormat {
+            /// Network/AddressType is reserved for future use.
+            pub fn is_reserved(&self) -> bool {
+                if let Ok(known) = KnownSs58AddressFormat::try_from(*self) {
+                    matches!(known, #(KnownSs58AddressFormat::#reserved_identifiers)|*)
+                } else {
+                    false
+                }
+            }
+        }
+    })
+}
+
+fn main() {
+    let out_dir = env::var_os("OUT_DIR").expect("OUT_DIR should exist");
+
+    let result: String = match create_ss58_registry(include_str!("ss58-registry.json")) {
+        Ok(result) => result.to_string(),
+        Err(msg) => panic!("{}", msg),
+    };
+
+    let dest_path = Path::new(&out_dir).join("account_type_enum.rs");
+    fs::write(&dest_path, result).unwrap_or_else(|_| panic!("failed to write to {:?}", &dest_path));
```

Comment:
Use an expect here. It would be good to know why it failed.

Reply:
Will add a `#![allow(clippy::expect_fun_call)]` at the top of build.rs to tell it to allow it.

Reply:
Ohh fucking clippy.... Yeah that is a really expensive call...

Please remove this `allow`. I wanted to have the error printed. You can use `unwrap_or_else` and just add the error.

Reply:
Went for eprintln and a -1 exit code. Is that fine?